### PR TITLE
Add tests to ensure we can't add empty dashboard filter

### LINKED ISSUE #15462: It is possible to add an empty dashboard filter (contains)
**Describe the bug**
When using new dashboard filters, it is possible to set empty value.

**Logs**
N/A

**To Reproduce**
1. Go to dashboard with at least one card on it
2. Click on pencil icon to edit the dashboard
3. Click on filter icon to add a filter to the dashboard
4. Choose "Other categories" > Contains
5. Connect it to a column in the question on a dashboard
6. Save the dashboard
7. Click on that filter widget - you can type and "Add filter" button is correctly disabled
8. Enter a letter or two - observe that the list of values appear and "Add filter" button is now enabled
9. Delete what you typed
10. "Add filter" stays enabled and you can now click on it adding an empty filter value

**Expected behavior**
"Add filter" button should go back to disabled state when there is no input value, or when it is deleted.

**Screenshots**

**Information about your Metabase Installation:**
- v0.39.0-SNAPSHOT
- Built on 2021-04-01src_hash=a6e9f576a52174cff5ecaa741cd4fed22cbae3e9
- Branch: master
- Hash: dfa9398

**Severity**
Low.

### PR DESCRIPTION
**Pull request by WiNloSt**
Closes https://github.com/metabase/metabase/issues/15462

### Description

The issue has already been fixed. [This is confirmed](https://metaboat.slack.com/archives/C057LE0Q4PQ/p1689953430097339?thread_ts=1689865605.974619&cid=C057LE0Q4PQ). This PR just adds tests to ensure that.

### How to verify

Follow the repro in https://github.com/metabase/metabase/issues/15462. Actually, this issue has already been fixed. I don't know since when though.

### Checklist

- [x] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

File: frontend/src/metabase/parameters/components/ParameterValueWidget.css
Comment (WiNloSt) [inline frontend/src/metabase/parameters/components/ParameterValueWidget.css:1]: I only remove the styles that are used in `ParameterWidget`. Previously, this file is used in both
- `ParameterWidget` and
- `ParameterValueWidget`

File: frontend/src/metabase/parameters/components/ParameterValueWidget.jsx [hunk 1]
```diff
@@ -33,7 +33,7 @@ import {
 } from "metabase-lib/parameters/utils/parameter-type";
 
 import ParameterFieldWidget from "./widgets/ParameterFieldWidget/ParameterFieldWidget";
-import S from "./ParameterWidget.css";
+import S from "./ParameterValueWidget.css";
```
Comment (WiNloSt) [inline frontend/src/metabase/parameters/components/ParameterValueWidget.jsx:36]: Rename the style file since most of the style rules are actually used by `ParameterValueWidget`

File: frontend/src/metabase/parameters/components/ParameterWidget/ParameterWidget.jsx [hunk 2]
```diff
@@ -4,12 +4,11 @@ import PropTypes from "prop-types";
 import cx from "classnames";
 import { Icon } from "metabase/core/components/Icon";
 import { color } from "metabase/lib/colors";
-import FieldSet from "../../components/FieldSet";
-import ParameterValueWidget from "./ParameterValueWidget";
+import ParameterValueWidget from "../ParameterValueWidget";
 
-import S from "./ParameterWidget.css";
+import { StyledFieldSet } from "./ParameterWidget.styled";
```
Comment (WiNloSt) [inline frontend/src/metabase/parameters/components/ParameterWidget/ParameterWidget.jsx:9]: Remove CSS module

File: frontend/src/metabase/parameters/components/ParameterWidget/ParameterWidget.styled.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import styled from "@emotion/styled";
+import FieldSet from "metabase/components/FieldSet";
+import { color } from "metabase/lib/colors";
+
+export const StyledFieldSet = styled(FieldSet)`
```
Comment (WiNloSt) [inline frontend/src/metabase/parameters/components/ParameterWidget/ParameterWidget.styled.ts:5]: This is the same style previously defined in `frontend/src/metabase/parameters/components/ParameterWidget.css`. Nothing has been changed.

File: frontend/src/metabase/parameters/components/ParameterValueWidget.css [hunk 4]
```diff
@@ -1,28 +1,3 @@
...
-  margin-left: -0.45em;
-  padding: 0 0.5em;
-}
-
-:local(.container.deemphasized) {
-  opacity: 0.4;
-}
-:local(.container.deemphasized:hover) {
-  opacity: 1;
-}
```
Comment (WiNloSt) [inline frontend/src/metabase/parameters/components/ParameterValueWidget.css:24]: Please note that these rules aren't used, so I didn't migrate these rules.

File: frontend/src/metabase/parameters/components/ParametersList.jsx [hunk 5]
```diff
@@ -8,7 +8,7 @@ import {
   SortableHandle,
 } from "metabase/components/sortable";
 import { getVisibleParameters } from "metabase/parameters/utils/ui";
-import StaticParameterWidget from "./ParameterWidget";
+import { ParameterWidget as StaticParameterWidget } from "./ParameterWidget";
```
Comment (WiNloSt) [inline frontend/src/metabase/parameters/components/ParametersList.jsx:11]: Use name import as default import are discouraged.

File: frontend/src/metabase/parameters/components/ParameterWidget/ParameterWidget.unit.spec.tsx [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+    type: "text/contains",
+    slug: "text_contains",
+    name: "Text contains",
+  });
+
+  const setValue = jest.fn();
+
+  renderWithProviders(
+    <ParameterWidget
+      // parameters={[parameter]}
```
[the remark this answers is not in the document]
Comment (WiNloSt) [reply frontend/src/metabase/parameters/components/ParameterWidget/ParameterWidget.unit.spec.tsx:20]: Good catch, I should remove that. I was adding that line thinking if it will fix the failed test while I was setting up all the tests.

File: frontend/src/metabase/parameters/components/ParameterWidget/ParameterWidget.unit.spec.tsx [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+    expect(screen.getByRole("button", { name: "Add filter" })).toBeEnabled();
+
+    userEvent.type(textInput, "{backspace}".repeat(text.length));
+    expect(screen.getByRole("button", { name: "Add filter" })).toBeDisabled();
+
+    userEvent.click(screen.getByRole("button", { name: "Add filter" }));
+    expect(setValue).not.toHaveBeenCalled();
+  });
+
+  it("should not be able to submit empty value when parameter has connected field (metabase#15462)", () => {
```
Comment (WiNloSt) [inline frontend/src/metabase/parameters/components/ParameterWidget/ParameterWidget.unit.spec.tsx:67]: This is actually the repro for the issue #15462

**Review by uladzimirdev — APPROVED** [review]
LGTM
